src/cogent3/core/alignment.py

```diff
@@ -564,6 +564,11 @@ def __init__(
         if isinstance(data, (typing.Generator, typing.Iterator)):
             data = tuple(data)
 
+        if len(data) == 0:
```

Comment:
I assume the conversion above ensures that data has a `__len__` attribute, though I note that the coercion in `conversion_f` checks again for whether it is a Generator/Iterator.

Reply:
Correct and the redundancy in this code will be addressed with the "big rewrite" of backends to sequence collections.